crates/oxc_linter/src/rules/typescript/prefer_for_of.rs

```diff
@@ -0,0 +1,296 @@
+                (
+                    SimpleAssignmentTarget::AssignmentTargetIdentifier(id),
+                    UpdateOperator::Increment,
+                ) => id.name == var_name,
+                _ => false,
+            },
+            Expression::AssignmentExpression(expr) => {
+                if !matches!(&expr.left,
+                    AssignmentTarget::SimpleAssignmentTarget(
+                        SimpleAssignmentTarget::AssignmentTargetIdentifier(id)
+                    )
+                    if id.name == var_name
+                ) {
+                    return false;
+                }
+
+                match expr.operator {
+                    AssignmentOperator::Addition => {
+                        matches!(&expr.right, Expression::NumericLiteral(lit)
+                            if (lit.value - 1f64).abs() < f64::EPSILON)
+                    }
+                    AssignmentOperator::Assign => {
+                        let Expression::BinaryExpression(bin_expr) = &expr.right else {
+                            return false;
+                        };
+
+                        if bin_expr.operator != BinaryOperator::Addition {
+                            return false;
+                        }
+
+                        match (&bin_expr.left, &bin_expr.right) {
+                            (Expression::Identifier(id), Expression::NumericLiteral(lit))
+                            | (Expression::NumericLiteral(lit), Expression::Identifier(id)) => {
+                                id.name == var_name && (lit.value - 1f64).abs() < f64::EPSILON
+                            }
+                            _ => false,
+                        }
+                    }
+                    _ => false,
+                }
+            }
+            _ => false,
+        }
+    }
+}
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("typescript-eslint(prefer-for-of): Expected a `for-of` loop instead of a `for` loop with this simple iteration.")]
+#[diagnostic(severity(warning), help("Consider using a for-of loop for this simple iteration."))]
+struct PreferForOfDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct PreferForOf;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Enforces the use of for-of loop instead of a for loop with a simple iteration.
+    ///
+    /// ### Why is this bad?
+    /// Using a for loop with a simple iteration over an array can be replaced with a more concise
+    /// and readable for-of loop. For-of loops are easier to read and less error-prone, as they
+    /// eliminate the need for an index variable and manual array access.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // Bad
+    /// for (let i = 0; i < arr.length; i++) {
+    ///   console.log(arr[i]);
+    /// }
+    ///
+    /// // Good
+    /// for (const item of arr) {
+    ///   console.log(item);
+    /// }
+    /// ```
+    PreferForOf,
+    correctness
```

Comment:
Could you change it to [`style`](https://github.com/oxc-project/oxc/blob/4c5abb590ec214023c3ec5d43616afd8a85b5f6e/crates/oxc_cli/src/command/lint.rs#L83)?  It looks like the `prefer-*` rule more considered to be best practice.

Reply:
Done.